templates/ast-iam-roles.template.yaml

```diff
@@ -238,13 +238,15 @@ Resources:
                   - arn:${AWS::Partition}:s3:::${S3Bucket}/${QSS3KeyPrefix}*
                   - S3Bucket: !If [UsingDefaultBucket, !Sub '${QSS3BucketName}-${AWS::Region}', !Ref QSS3BucketName]
                 Effect: Allow
-  CognitoInspectRole:
+  CognitoAdminRole:
     Type: AWS::IAM::Role
     Properties:
-      RoleName: !Sub ${ProjectName}-CognitoInspectRole
+      RoleName: !Sub ${ProjectName}-CognitoAdminRole
       AssumeRolePolicyDocument:
         Statement:
-          - Action: sts:AssumeRole
+          - Action:
+              - cognito-idp:DescribeUserPoolClient
+              - cognito-idp:AdminCreateUser
```

Comment:
These actions need to be under the main, inline policy, not the AssumeRolePolicyDocument

Reply:
Nice catch thanks!